source/LocationFilter.cpp

```diff
@@ -180,6 +206,9 @@ bool LocationFilter::Matches(const Planet *planet, const System *origin) const
 	if(!governments.empty() && !governments.count(planet->GetGovernment()))
 		return false;
 	
+	if(!MatchesNeighborFilters(neighborFilters, planet->GetSystem(), origin))
+		return false;
```

Comment:
Is this necessary here? It seems to me that neighbor filters will only ever apply to systems, not to planets, so it's sufficient for Matches(System...) to check the neighbors.

Reply:
I'll investigate, but I think this is needed in case the neighbor filter is being checked a a part of a source, destination, or stopover filter.

Reply:
You should be able to just modify Matches(System...) so that it checks the neighbor filter whether or not `didPlanet` is true. The only checks in that function that are gated behind `!didPlanet` are the ones that could apply either to a planet or to a system.